apps/user/src/hocs/withAuth.tsx

```diff
@@ -27,24 +27,31 @@ const withAuth = (Component: React.ComponentType) => {
         setIsMounted(true);
       }
     }, []);
+useEffect(() => {
+  if (isMounted) {
+    if (!now.isBetween(제출_시작_날짜, 제출_마감_날짜)) {
+      overlay.open(({ isOpen, close }) => (
+        <GuardFormModal isOpen={isOpen} onClose={close} />
+      ));
+      return;
+    } else if (LOGIN_TYPE === 'ADMIN') {
+      alert('권한이 없어 원서 작성이 불가합니다.');
+      router.replace(ROUTES.MAIN);
+      return;
+    } else if (!hasAccessToken) {
+      overlay.open(({ isOpen, close }) => (
+        <NeedLoginModal isOpen={isOpen} onClose={close} />
+      ));
+      return;
+    } else if (LOGIN_TYPE === undefined || LOGIN_TYPE === null) {
```

Comment:
```suggestion
    } else if (LOGIN_TYPE == undefined || LOGIN_TYPE === null) {
```

https://www.slash.page/ko/libraries/common/utils/src/isNil.i18n

요론 라이브러리로 추상화가 가능할 것 같아요~ 사실 이건 그냥 참고용으로 이런 것도 있다~~라고만 알아놔주세요 굳이 바꿀 필요는 없어보여요